### ISSUE TITLE
Expose plugin initialization method for Dynamically Imported Modules

### ISSUE THREAD

**Issue by csrcreative** [zurb/foundation-sites#11291]
This may be a bit of a stretch but I figured I'd just put this out there and see if there is any desire in the community for this feature.

I am currently using Webpack 4 dynamic import feature to load Foundation (6.4.1) plugins to the page on interaction with the component. The issue I have run into is when initializing the component with the necessary aria attributes, before loading the plugin js. I am currently duplicating much of what is happening in the plugins init method for adding classes and aria attributes before actually making a request for the plugin js itself.

Ideally the Foundation component would have all the necessary aria attributes on page load, so even if a user never interacted with the component, it would still have all the semantic goodness attached to it for users better understanding of the purpose of said component.

Here is an example for the Accordion menu:
```javascript

import { setAttributes } from "../utilities";

let loadAccordionMenu = () => {
    let component = $(".accordion-menu");
    let accordionMenu = document.querySelectorAll(".accordion-menu").length
        ? document.querySelectorAll(".accordion-menu")
        : false;

    if (accordionMenu != false) {

        let _init = () => {
            _accessibilityAttributes();

            document.body.addEventListener("click", _clickHandler);
        };

        //Setting attributes, this is taken directly from the accordionmenu plugin _init method.
        let _accessibilityAttributes = () => {
            //The Foundation.Nest utility adds classes to nested menus
            Foundation.Nest.Feather(component, "accordion");

            //TODO: ACCOUNT FOR MULTIPLE MENUS
            setAttributes(accordionMenu[0], {
                role: "tree",
                "aria-multiselectable": true,
                "data-accordion-menu": Foundation.GetYoDigits(
                    6,
                    "accordion-menu"
                )
            });

            let menuLinks = accordionMenu[0].querySelectorAll(
                ".is-accordion-submenu-parent"
            );

            menuLinks.forEach(function(element, key) {
                let linkId = Foundation.GetYoDigits(6, "acc-menu-link"),
                    sub = element.querySelector("[data-submenu]"),
                    subId = Foundation.GetYoDigits(6, "acc-menu");

                setAttributes(element, {
                    "aria-controls": subId,
                    "aria-expanded": false,
                    id: linkId
                });

                //TODO: ADD AN OPTION TO DISPLAY SOME MENUS AS ACTIVE
                setAttributes(sub, {
                    "aria-labelledby": linkId,
                    "aria-hidden": true,
                    role: "group",
                    id: subId,
                    style: "display:none"
                });
            });
        };

        let _clickHandler = e => {
            if (e.target.closest(".accordion-menu")) {
                _importPlugin(() => {
                    //initializing the plugin
                    component.foundation();

                    //Initiate the first open
                    component.foundation(
                        "toggle",
                        $(e.target.nextElementSibling)
                    );
                    //remove the event listener.
                    document.body.removeEventListener("click", _clickHandler);
                });
            }
        };
        let _importPlugin = callback => {
            import(/* webpackChunkName: "accordionmenu" */ `foundation-sites/js/foundation.accordionMenu.js`).then(
                plugin => {
                    Foundation.plugin(plugin.AccordionMenu, "AccordionMenu");
                    callback();
                }
            );
        };

        _init();
    }
};
export default loadAccordionMenu;

```

Would there be a way to somehow make a plugin's init method accessible without wholesale importing the entire plugin itself?

**Comment by ncoden** [comment]
Hi @csrcreative,

I don't really understand the issue. Most (if not all) components get their a11y attributes added at initialization. On page load if you called `$(document).foundation()` or when you dynamically instantiate them with `new Foundation.Accordion($elem)`.

I don't see why you would have to copy internal code or "get a better access to init method". You can initialize the plugin by yourself when its HTML is ready.

Is there anything I got wrong ?

**Comment by csrcreative** [comment]
Ok, so it may be my lack of understanding that’s to blame here, and I apologize if that’s the case.

My intention here is to load these plugins “on demand”.

On page load I serve up a single bundled js file that has all the required dependencies for the Foundation plugins to work (i.e. JQuery, the core file, utilities, etc) but NO actual plugins. I also include what I am calling “plugin loader” modules. These basically check for the existence of a given plugin and then create event listeners that will load the plugin if a user interacts with it. 

So if you were to go to a page with an Orbit component on it, the required plugin js isn’t loaded on to the page until a user interacts with it. However, I still need the Orbit component to have all the necessary classes and attributes attached, even if the user never actually interacts with it.

Hence my current approach of duplicating the initialization. If there is currently a way to initialize a component without actually including the plugin js I would of course pursue that.

**Comment by ncoden** [comment]
Hmmm I understand the use case but...

> If there is currently a way to initialize a component without actually including the plugin js I would of course pursue that.

...is a weird request and I doubt that any library will be able to fulfil it. Plugin instance initialization is part of each plugin code. You can optimize your page to only load the JS for the plugins that are in the page, but from there I would recommand you to initialize all components (so the a11y attributes are set) without waiting for the user to interact with them.

**Comment by csrcreative** [comment]
Yep, I see your point, and I may need to take another look at my approach. Thanks for the quick response.

**Comment by ncoden** [comment]
Glap to help. I'm closing this issue, feel free to reopen if you still have an issue with your new approach.
